ui/apps/platform/src/hooks/useURLParameter.ts

```diff
@@ -36,23 +36,23 @@ function useURLParameter(keyPrefix: string, defaultValue: QueryValue): UseURLPar
     // as the URL parameters do not change
     const setValue = useCallback(
         (newValue: QueryValue, historyAction: Action = 'push') => {
-            const previousQuery = getQueryObject(location.search) || {};
-            const newQueryString = getQueryString({
+            const previousQuery = getQueryObject(history.location.search) || {};
```

Comment:
According to the react router docs, `history.location` is mutable while `location` is mutable, and they generally recommend the latter because of this. However in this case we want to rely on the mutability in order to get the latest version of `location.search` when this function is called multiple times from application code.

Reply:
You are super searcher! Even more important, is knowing when you need to ;)

Please write a comment which summarizes what what you learned, maybe even with link to the docs, so we can adjust if needed when we replace `useHistory` with `useNavigate` in React Router 6.

Reply:
@dvail I assume you meant to say `location` is immutable which is why it's recommended?

Reply:
Whoops, yeah `location` is immutable, I'll edit.

Also Mark, I believe React Router 6 has a `useParams` hook or similar which might even render this function obsolete.